### ISSUE TITLE
selenium.common.exceptions.WebDriverException on trying to run tests in Firefox under python3.11.2

### ISSUE THREAD

**Issue by olait2018** [seleniumbase/SeleniumBase#1813]
Hello,

Recently I've upgraded python3.8 to 3.11.2 and tried to run tests in Firefox, but receive following error:
...

selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1
...

**Environment**:
```
# uname -a
Linux fba36d108b02 5.10.76-linuxkit #1 SMP PREEMPT Mon Nov 8 11:22:26 UTC 2021 x86_64 GNU/Linux

# python3 --version
Python 3.11.2

# pytest --version
pytest 7.2.2

# firefox --full-version
Mozilla Firefox 111.0 20230309232128 20230309232128

# geckodriver --version
geckodriver 0.32.0 (4563dd583110 2022-10-13 09:22 +0000)
```
NOTE: I tried also [latest](https://github.com/mozilla/geckodriver/releases#:~:text=0.32.2-,Latest,-0.32.2%20(2023%2D02) driver - 0.32.2, but no luck.

```
# pip show seleniumbase
Name: seleniumbase
Version: 4.13.15

# echo $PATH
/automation_web/new_venv/bin:/usr/local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/lib/jvm/java-11-openjdk-amd64/bin:/opt/firefox:/usr/local/bin/geckodriver
```

I tried to run [test_basic](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/basic_test.py) with the following command:
`pytest -k test_basics --headless --browser=firefox `
, but receive long stack-trace with error: 
``` 
...
self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x400dbd2a90>, response = {'status': 500, 'value': '{"value":{"error":"unknown error","message":"Process unexpectedly closed with status 1","stacktrace":""}}'}
...
new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py:245: WebDriverException 
FAILED basic_test.py::MyTestClass::test_basics - selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1
```

Full error log is in the file [full_error_log.txt](https://github.com/seleniumbase/SeleniumBase/files/11023596/full_error_log.txt)

I tried to verify geckodriver as is described [here](https://seleniumbase.io/help_docs/verify_webdriver/#verifying-geckodriver-firefox-webdriver:~:text=Verifying%20Geckodriver%20(Firefox%20WebDriver)) , but with the only difference - headless=True , and received following error:
```
# python3
Python 3.11.2 (main, Mar 17 2023, 02:28:16) [GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from seleniumbase import get_driver
>>> driver = get_driver("firefox", headless=True)
Traceback (most recent call last):
  File "/automation_web/new_venv/lib/python3.11/site-packages/seleniumbase/core/browser_launcher.py", line 1958, in get_local_driver
    return webdriver.Firefox(
           ^^^^^^^^^^^^^^^^^^
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/firefox/webdriver.py", line 196, in __init__
    super().__init__(command_executor=executor, options=options, keep_alive=True)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 286, in __init__
    self.start_session(capabilities, browser_profile)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 378, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/automation_web/new_venv/lib/python3.11/site-packages/seleniumbase/core/browser_launcher.py", line 1236, in get_driver
    return get_local_driver(
           ^^^^^^^^^^^^^^^^^
  File "/automation_web/new_venv/lib/python3.11/site-packages/seleniumbase/core/browser_launcher.py", line 1979, in get_local_driver
    return webdriver.Firefox(
           ^^^^^^^^^^^^^^^^^^
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/firefox/webdriver.py", line 196, in __init__
    super().__init__(command_executor=executor, options=options, keep_alive=True)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 286, in __init__
    self.start_session(capabilities, browser_profile)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 378, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "/automation_web/new_venv/lib/python3.11/site-packages/selenium/webdriver/remote/errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1
>>> 
```

Could you please help to resolve this issue?
Thanks in advance!

**Comment by olait2018** [comment]
In the same time simple pure selenium test works ok in the same environment:
```
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.options import Options
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager


options = Options()
options.add_argument('-headless')
options.add_argument('--window-size=1920,1080')
options.add_argument('start-maximized')
driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()), options=options)

driver.get("https://www.selenium.dev/")

# wait for page to load
element = WebDriverWait(driver=driver, timeout=5).until(
    EC.presence_of_element_located((By.CSS_SELECTOR, '.h3.mb-3.selenium-webdriver'))
)
assert driver.find_element(By.CSS_SELECTOR, ".h4.mb-3.selenium-webdriver").is_displayed(), "'.h3.mb-3.selenium-webdriver' not found"

driver.close()
driver.quit()
```

**Comment by mdmintz** [comment]
Try it with one of the 23 Syntax Formats: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md

You used:

```python
from seleniumbase import get_driver
driver = get_driver("firefox", headless=True)
```

But you might want to try this instead:

```python
from seleniumbase import Driver
driver = Driver(browser="firefox", headless=True)
```

Your way was calling ``browser_launcher.py`` directly, which is missing key steps.

**Comment by olait2018** [comment]
thanks @mdmintz 
I've tried, but no luck, I still get error:
```
...
selenium.common.exceptions.WebDriverException: Message: Process unexpectedly closed with status 1
...
```

Full log and list of commands I executed is here 
[logs.txt](https://github.com/seleniumbase/SeleniumBase/files/11026267/logs.txt)

**Comment by olait2018** [comment]
many thanks, @mdmintz 
Workaround works good!

**Comment by mdmintz** [comment]
Resolved in ``4.13.16`` - https://github.com/seleniumbase/SeleniumBase/releases/tag/v4.13.16 without need of a workaround. If any issues, let me know.
